### ISSUE TITLE
_Widget.cancel() doesnt cancel

### ISSUE THREAD

**Issue by martinohanlon** [lawsie/guizero#74]
_Widget.cancel() didn't cancel an after call. 

I'll commit a change to the 0.4 branch to resolve.

**Comment by martinohanlon** [comment]
I refactored the after, repeat, cancel and call_wrapper functions to resolve this issue. 

I have also include an example of how to use them.

[commit](https://github.com/lawsie/guizero/commit/eec02a6777d353da76cba001dbc3cab000d6112a)

**Comment by martinohanlon** [comment]
@lawsie as this is my first commit I thought I had better show 'full working' and give you a chance to comment / redo / undo

**Comment by martinohanlon** [comment]
Added functionality to allows arguments to be passed to after and repeat functions [commit](https://github.com/lawsie/guizero/commit/a9c0037e8a2814a9adb69326ea4f666683004773#diff-d81ef841c670edfe499c68c98716ccc2L13)

**Comment by lawsie** [comment]
Great. At the moment if you want to specify a command for something (e.g. a PushButton) with arguments, you have to specify the arguments as a list e.g.

`button = PushButton(app, my_function, args=[1, 2])`

How would specifying arguments to after/repeat work with this code, as I suspect it's different? Should this be changed in the other classes? This seems like probably a better way, but for backwards compatibility with older versions the old way will have to stay (as well) for now?

https://github.com/lawsie/guizero/blob/master/guizero/PushButton.py

**Comment by martinohanlon** [comment]
If you wanted to call my_function on click of a button but also have it repeat every 1000 milliseconds, you would use:

```
def my_function(arg1, arg2):
    print(arg1)
    print(arg2)

button = PushButton(app, my_function, args=[1, 2])

button.repeat(1000, my_function, 3, 4)
```

I followed tkinter's implementation of after(), which uses *args for any arguments, but it could be very easily changed to:

`button.repeat(1000, my_function, args = [3,4]`

... this feel less 'simple' but more consistent with widgets.

Thoughts?

**Comment by Coal0** [comment]
The latest commit takes `*args` arguments. If backward compatibility is of great concern, this could be changed to `args=[]` (in which case I'd also allow keyword arguments, i.e. `kwargs={}`).

**Comment by lawsie** [comment]
Yeah, this is where you are exposed to me writing guizero as a very steep learning curve from Jon Snow to...Jon Snow version 2 :D Annoyingly your way is better, but I didn't know about that at the time. 

I think we should keep the slightly crappy `args=` syntax for backwards compatibility, but if there's also the opportunity to add the new and better way as an alternative then we can warn about it in the docs/in a warning and then remove the crappy way in version 1.

**Comment by martinohanlon** [comment]
I'll change to args = []

**Comment by martinohanlon** [comment]
Updated to use args = []. 

[commit](https://github.com/lawsie/guizero/commit/11358b4717c17925d5f0120834bae2277b62ddd6)

**Comment by martinohanlon** [comment]
Closing, resolved in 0.4
